src/components/shared/DivWithTooltip.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import Tooltip from './Tooltip';
+import type { CssPixels } from '../../types/units';
+
+type Props = {
+  tooltip: React.Node,
+  children?: React.Node,
+};
+
+type State = {|
+  isMouseOver: boolean,
+  mouseX: CssPixels,
+  mouseY: CssPixels,
+|};
+
+/**
+ * This component provides a way to automatically insert a tooltip when mousing over
+ * a div.
+ */
+export default class DivWithTooltip extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isMouseOver: false,
+      mouseX: 0,
+      mouseY: 0,
+    };
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this._onMouseMove, false);
+  }
+
+  _onMouseEnter = () => {
+    this.setState({ isMouseOver: true });
+    document.addEventListener('mousemove', this._onMouseMove, false);
+  };
+
+  _onMouseLeave = () => {
+    this.setState({ isMouseOver: false });
+    document.removeEventListener('mousemove', this._onMouseMove, false);
+  };
+
+  _onMouseMove = (event: MouseEvent) => {
+    this.setState({
+      mouseX: event.pageX,
+      mouseY: event.pageY,
+    });
+  };
+
+  render() {
+    const { mouseX, mouseY, isMouseOver } = this.state;
+    const { children, tooltip } = this.props;
+    const shouldShowTooltip = isMouseOver;
+
+    // Pass through the props without the tooltip property.
+    const containerProps = Object.assign({}, this.props);
+    delete containerProps.tooltip;
```

Comment:
FYI with https://babeljs.io/docs/plugins/transform-object-rest-spread/ (stage 3, not enabled in our config but we could), this could be written like this:
```javascript
const { tooltip, ...containerProps } = this.props;
const { children } = containerProps; // too bad we can't do it in one go :)
```
And I even think Flow understands it better.

I'd be fine adding this babel transform if you think it's better.